mParticle-Apple-SDK/Listener/MPListenerController.swift

```diff
@@ -0,0 +1,353 @@
+//
+//  MPListenerController.swift
+//  mParticle-Apple-SDK
+//
+//  Created by Brandon Stalnaker on 12/16/24.
+//
+
+import Foundation
+
+@objc public enum MPEndpoint: Int {
+    case identityLogin = 0, identityLogout, identityIdentify, identityModify, events, config, alias
+}
+@objc public enum MPDatabaseTable: Int {
+    case attributes = 0, breadcrumbs, messages, reporting, sessions, uploads, unknown
+}
+
+@objc public protocol MPListenerProtocol {
+
+    /**
+     * Indicates that an API method was called. This includes invocations both from external sources (your code)
+     * and those which originated from within the SDK
+     * @param apiName the name of the API method
+     * @param stackTrace is the current stackTrace as an array of NSStrings
+     * @param isExternal true, if the call originated from outside of the SDK
+     * @param objects is the arguments sent to this api, such as the MPEvent in logEvent
+     */
+    @objc optional func onAPICalled(_ apiName: Any!, stackTrace: Any!, isExternal: Any!, objects: Any!)
```

Comment:
I assume the `!`'s here are to keep the same public interface as before?

Reply:
yep line 43 of mParticle-Apple-SDK/Include/MPListenerProtocol.h